File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderTrustAccountTest.java
package uk.gov.justice.hmpps.prison.repository.jpa.model;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class OffenderTrustAccountTest {
    @Test
    void isAccountClosed_NotSet() {
        assertThat(new OffenderTrustAccount().isAccountClosed()).isFalse();
    }

    @Test
    void isAccountClosed_Y_is_true() {
        final var offenderTrustAccount = new OffenderTrustAccount();
        offenderTrustAccount.setAccountClosedFlag("Y");
        assertThat(offenderTrustAccount.isAccountClosed()).isTrue();
    }

    @Test
    void isAccountClosed_N_is_false() {
        final var offenderTrustAccount = new OffenderTrustAccount();
        offenderTrustAccount.setAccountClosedFlag("N");
        assertThat(offenderTrustAccount.isAccountClosed()).isFalse();
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderTrustAccountTest.kt
package uk.gov.justice.hmpps.prison.repository.jpa.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class OffenderTrustAccountTest {
  @Test
  fun isAccountClosed_Y_is_true() {
    val offenderTrustAccount = OffenderTrustAccount(OffenderTrustAccountId("ASI", 1), true)
    assertThat(offenderTrustAccount.accountClosed).isTrue()
  }

  @Test
  fun isAccountClosed_N_is_false() {
    val offenderTrustAccount = OffenderTrustAccount(OffenderTrustAccountId("ASI", 1), false)
    assertThat(offenderTrustAccount.accountClosed).isFalse()
  }
}
